zkevm/src/circuit.rs

```diff
@@ -0,0 +1,176 @@
+use bus_mapping::circuit_input_builder::{Block as cBlock, CircuitInputBuilder};
+use bus_mapping::operation::OperationContainer;
+use bus_mapping::state_db::{Account, CodeDB, StateDB};
+use eth_types::{evm_types::OpcodeId, Field};
+use ethers_core::types::Bytes;
+use halo2_proofs::arithmetic::BaseExt;
+use halo2_proofs::plonk::Circuit;
+use is_even::IsEven;
+use pairing::bn256::Fr;
+use std::collections::HashMap;
+use types::eth::{AccountProofWrapper, BlockResult};
+use zkevm_circuits::evm_circuit::param::STEP_HEIGHT;
+use zkevm_circuits::evm_circuit::table::FixedTableTag;
+use zkevm_circuits::evm_circuit::test::TestCircuit;
+use zkevm_circuits::evm_circuit::witness::{block_convert, Block, RwMap};
+use zkevm_circuits::state_circuit::StateCircuit;
+
+pub static DEGREE: usize = 18;
+
+/// For keygen_vk.
+pub fn create_state_circuit() -> StateCircuit<Fr> {
+    let rw_map = RwMap::from(&OperationContainer {
+        memory: vec![],
+        stack: vec![],
+        storage: vec![],
+        ..Default::default()
+    });
+
+    StateCircuit::<Fr>::new(Fr::rand(), rw_map)
+}
+
+pub fn block_result_to_circuits<F: Field>(
```

Comment:
Again, why not just a `From` impl?

Reply:
This function is `BlockResult -> Result<(Circuit, Circuit)> `,  I think `From trait` is not satisfied.